File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.cpp
class Solution {
public:
    int arithmeticTriplets(vector<int>& nums, int diff) {
        // int i = 0, j = 1, k = 2, count = 0, n = nums.size();
        // for(int i = 0; i < n - 2; i++) {
        //     for(int j = i + 1; j < n -1; j++) {
        //         for(int k = j + 1; k < n; k++) {
        //             if(((nums[j] - nums[i]) == diff) && ((nums[k] - nums[j]) == diff)) {
        //                 count++;
        //             }
        //         }
        //     }
        // }
        

        // int count = 0;
        // unordered_map<int, bool> m = {};
        // for(int i = 0; i < nums.size(); i++) {
        //     m[nums[i]] = true;
        // }
        // for(int i = 0; i < nums.size(); i++) {
        //     count += m[nums[i] - diff] && m[nums[i] + diff];
        // }
        // return count;
        
        
        int count = 0;
        unordered_set<int> s = {};
        for(int i = 0; i < nums.size(); i++) {
            if ( (s.find(nums[i] - diff) != s.end()) && (s.find(nums[i] - (2*diff)) != s.end()) ) {
                count++;
            }
            s.insert(nums[i]);
        }
        return count;
    }
};

File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.py
class Solution(object):
    def arithmeticTriplets(self, nums, diff):
        """
        :type nums: List[int]
        :type diff: int
        :rtype: int
        """
        
        # count = 0
        # s = set(nums)
        #
        # for num in nums:
        #     if ((num - diff) in s) and ((num + diff) in s):
        #         count += 1
        #
        # return count
        
        count = 0
        s = set()
        
        for num in nums:
            if ((num - diff) in s) and ((num - 2*diff) in s):
                count += 1
            s.add(num)
        
        return count
        